src/main/java/hudson/plugins/git/UserRemoteConfig.java

```diff
@@ -212,6 +214,45 @@ public FormValidation doCheckUrl(@AncestorInPath Item item,
             return FormValidation.ok();
         }
 
+        /**
+         * A form validation logic as a method to check the specification of 'refSpec' and notify the user about
+         * illegal specs before applying the project configuration
+         * @param name Name of the remote repository
+         * @param url Repository URL
+         * @param value value of RefSpec
+         * @return FormValidation.ok() or FormValidation.error()
+         * @throws IllegalArgumentException
+         */
+        public FormValidation doCheckRefspec(@QueryParameter String name,
+                                             @QueryParameter String url,
+                                             @QueryParameter String value) throws IllegalArgumentException {
+
+            String refSpec = Util.fixEmptyAndTrim(value);
+
+            if(refSpec == null){
+                return FormValidation.error(Messages.UserRemoteConfig_CheckUrl_UrlIsNull());
```

Comment:
Isn't this a change in behavior?  I thought that the plugin previously allowed an empty refspec and treated it as though it were the default refspec.  Could you double check that previous versions did not treat empty refspecs as the default refspec?

Reply:
Yes you are correct, I just checked it. I am adding the default refspec in case of an empty refspec.